genesyscloud/architect_flow/resource_genesyscloud_flow.go

```diff
@@ -113,10 +113,31 @@ func updateFlow(ctx context.Context, d *schema.ResourceData, meta interface{}) d
 	}
 
 	s3Uploader := files.NewS3Uploader(reader, nil, substitutions, headers, "PUT", presignedUrl)
-	_, err = s3Uploader.Upload()
+	fileInfo, err := os.Stat(filePath)
 	if err != nil {
+		log.Printf("failed to read file information. Path: '%s' Error: %v", filePath, err)
+	}
+
+	uploadErr := util.WithRetries(ctx, 20*time.Second, func() *retry.RetryError {
+		uploadStartTime := time.Now()
+		_, err = s3Uploader.Upload()
+		uploadDuration := time.Since(uploadStartTime)
+		if err == nil {
+			return nil
+		}
+		// Retrying when encountering the error from DEVTOOLING-425
+		if strings.Contains(fmt.Sprintf("%v", err), "An existing connection was forcibly closed by the remote host") {
+			log.Printf("failed to upload file %s after %d nanoseconds (%v seconds). Error: %v", filePath, uploadDuration.Nanoseconds(), uploadDuration.Seconds(), err)
```

Comment:
If anything I would record milliseconds and seconds.  Not nanoseconds.  A network upload is almost never going to take nanoseconds.

Reply:
Done